src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,232 @@
+/-
+Copyright (c) 2022 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import linear_algebra.finite_dimensional
+
+/-!
+
+# Projective Spaces
+
+This file contains the definition of the projectivization of a vector space over a field,
+as well as the bijection between said projectivization and the collection of all one
+dimensional subspaces of the vector space.
+
+## Constructing terms of `projectivization K V`.
+We have three ways to construct terms of `projectivization K V`:
+- `projectivization.mk K v hv` where `v : V` and `hv : v ≠ 0`.
+- `projectivization.mk' K v` where `v : { w : V // w ≠ 0 }`.
+- `projectivization.mk'' H h` where `H : submodule K V` and `h : finrank H = 1`.
+
+## Other definitions
+- For `v : projectivization K V`, `v.submodule` gives the corresponding submodule of `V`.
+- `projectivization.equiv_submodule` is the equivalence between `projectivization K V`
+  and `{ H : submodule K V // finrank H = 1 }`.
+- For `v : projectivization K V`, `v.rep : V` is a representative of `v`.
+
+-/
+
+variables (K V : Type*) [field K] [add_comm_group V] [module K V]
+
+/-- The setoid whose quotient is the projectivization of `V`. -/
+def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
+{ r := λ u v, ∃ a : K, a • (u : V) = v,
```

Comment:
Is this different to the quotient used by `module.ray?

Reply:
It seems that `module.ray` is about positive rays, hence requires an ordering on the ring.

Reply:
Nice! In fact, following @riccardobrasca 's suggestion on zulip, I may just change the definition and use `mul_action.orbit_rel` with the units acting as the definition.

Reply:
Can you link the Zulip thread in the PR description? I hadn't seen it, but I agree with that idea (and elaborated on it below)

Reply:
I changed the definition to use multiplication by units directly, so this is no longer relevant.